### ISSUE TITLE
NameError: name 'quant_cuda' is not defined

### ISSUE THREAD

**Issue by SupaHam** [jllllll/GPTQ-for-LLaMa-Wheels#1]
hey,

I've installed the latest wheel because I'm failing to install GPTQ-for-LLaMa. But now when I try to run the model and type my first prompt i get this error:

```
(textgen) PS C:\Users\supah\Downloads\LLaMA-HFv2\oobabooga\text-generation-webui> python .\server.py --cai-chat --model llama-7b-4bit --wbits 4
Loading llama-7b-4bit...
CUDA extension not installed.
Loading model ...
Done.
Loaded the model in 3.18 seconds.
Loading the extension "gallery"... Ok.
C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\gradio\deprecation.py:40: UserWarning: The 'type' parameter has been deprecated. Use the Number component instead.
  warnings.warn(value)
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Exception in thread Thread-3 (gentask):
Traceback (most recent call last):
  File "C:\Users\supah\miniconda3\envs\textgen\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\supah\miniconda3\envs\textgen\lib\threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\supah\Downloads\LLaMA-HFv2\oobabooga\text-generation-webui\modules\callbacks.py", line 63, in gentask
    ret = self.mfunc(callback=_callback, **self.kwargs)
  File "C:\Users\supah\Downloads\LLaMA-HFv2\oobabooga\text-generation-webui\modules\text_generation.py", line 218, in generate_with_callback
    shared.model.generate(**kwargs)
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\torch\autograd\grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\transformers\generation\utils.py", line 1462, in generate
    return self.sample(
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\transformers\generation\utils.py", line 2478, in sample
    outputs = self(
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\transformers\models\llama\modeling_llama.py", line 765, in forward
    outputs = self.model(
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\transformers\models\llama\modeling_llama.py", line 614, in forward
    layer_outputs = decoder_layer(
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\transformers\models\llama\modeling_llama.py", line 309, in forward
    hidden_states, self_attn_weights, present_key_value = self.self_attn(
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\transformers\models\llama\modeling_llama.py", line 209, in forward
    query_states = self.q_proj(hidden_states).view(bsz, q_len, self.num_heads, self.head_dim).transpose(1, 2)
  File "C:\Users\supah\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\supah\Downloads\LLaMA-HFv2\oobabooga\text-generation-webui\repositories\GPTQ-for-LLaMa\quant.py", line 282, in forward
    quant_cuda.vecquant4matmul(x, self.qweight, y, self.scales, self.qzeros, self.groupsize)
NameError: name 'quant_cuda' is not defined
```

Do you know why this is happening?

**Comment by jllllll** [comment]
Make sure to install the wheel from inside of the GPTQ-for-LLaMa. I've had issues with it in the past when the wheel file was not in that folder. Also, make sure to uninstall whatever version of quant_cuda you may have before installing the wheel to insure that it actually gets installed.
`python -m pip uninstall quant_cuda`

Also make sure that the model that you are running is up-to-date. Older 4-bit models are not compatible with the latest GPTQ-for-LLaMa. https://github.com/oobabooga/text-generation-webui/wiki/LLaMA-model#step-2-get-the-pre-converted-weights

**Comment by SupaHam** [comment]
Hey thanks for getting back so quickly. I've tried the above, multiple times but to no avail. Actually updating to the new 4-bit models (and the repos) is what lead to me to create this issue. I have just started playing with this yesterday, and it seems a breakage or two happened as well. Before I was getting this output, which was complete gibberish, but it did seem to work fine ![image](https://user-images.githubusercontent.com/1299921/227810946-13e5be18-9893-4877-b4db-136b2f5efd3d.png)

**Comment by jllllll** [comment]
@SupaHam I've had that exact issue before. The cause for me was an incompatible/outdated wheel. What commands did you use to create your conda environment?

I've updated the wheel in this repo and added the command used to create the environment it was compiled in. You can also use this installer to automatically create the environment and setup the text-generation-webui: [Link](https://github.com/jllllll/one-click-installers)
I recently updated that installer as well.

**Comment by SupaHam** [comment]
Hey, sorry to keep you waiting - I had a bad internet connection. I managed to get it working creating a new fresh env and using your conda install. Some things were broken with chardet/other char packages. I ran `pip install cchardet` which fixed the issue for me and i was able to get the model to load and respond. It did seem to actually give me some responses, but they still come out a little buggy. I'm not sure if this is normal for the 7b model to be behaving like this. 
![image](https://user-images.githubusercontent.com/1299921/228377035-ff404b18-cd4f-4904-b8fb-66be7eeccfb4.png)

So, I went to try your one-click-installer and it works surprisingly well. Took about 25 mins to run the install script and had to modify the start script to use `--wbits=4` so i can load the 4bit weights. This is the output which is much better, but i'm guessing the `you:` in response is built into the model's behaviour?
![image](https://user-images.githubusercontent.com/1299921/228382092-b3facece-59e0-406d-a4f8-91e3ab4edc71.png)

Anywho, thank you very much for your help and work!
